Consider adding husky and lint-staged for precommit tasks

In the spirit of keeping code maintainable, pre-commit tasks are a blessing. Consider adding [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged) for managing pre-commit tasks.

Examples of pre-commit tasks can include running tests, linting and formatting. For a real world example of this in action, see https://github.com/forem/forem/blob/master/package.json#L28-L71 .

Here is an example where linting fails as well as tests for some changes that were made to a component.

```bash
husky > pre-commit (node v12.18.3)
  ✖ eslint --fix [FAILED]
    ◼ jest --findRelatedTests
  ↓ No staged files match {app,spec,config,lib}/**/*.rb [SKIPPED]
  ↓ No staged files match app/views/**/*.jbuilder [SKIPPED]
  ↓ No staged files match Gemfile [SKIPPED]
  ↓ No staged files match app/**/*.html.erb [SKIPPED]
  ↓ No staged files match *.json [SKIPPED]
  ↓ No staged files match *.scss [SKIPPED]
  ↓ No staged files match *.md [SKIPPED]
  ↓ No staged files match *.svg [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up... 

✖ eslint --fix:

/Users/nickytonline/dev/dev.to/app/javascript/crayons/Button/Button.jsx
  65:9  error  'ComponentName' is assigned a value but never used  no-unused-vars
  66:9  error  'Icon' is assigned a value but never used           no-unused-vars

✖ 2 problems (2 errors, 0 warnings)

husky > pre-commit hook failed (add --no-verify to bypass)
```

--- a/package.json
+++ b/package.json
@@ -41,6 +41,26 @@
     "watch:css": "postcss src/assets/tailwind.css -o src/assets/main.css",
     "cypress": "cypress open"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,jsx}": [
+      "prettier --write",
+      "eslint --fix"
+    ],
+    "*.json": [
+      "prettier --write"
+    ],
+    "*.md": [
+      "prettier --write --prose-wrap always"
+    ],
+    "*.css": [
+      "prettier --write"
+    ]
+  },
   "eslintConfig": {
     "extends": "react-app"
   },
@@ -68,6 +88,8 @@
     "eslint-plugin-prettier": "^3.1.3",
     "eslint-plugin-react": "^7.19.0",
     "eslint-plugin-react-hooks": "^2.5.0",
+    "husky": "^4.2.5",
+    "lint-staged": "^10.2.13",
     "node-fetch": "^2.6.0",
     "postcss-cli": "^7.1.0",
     "prettier": "^2.0.5",

